File: bin/bases.py
#!/usr/bin/env python3

import sys

def calculate_base_lengths(input_file, output_file="../../../CURRENT/bases.txt"):
    with open(input_file, "r") as infile, open(output_file, "w") as outfile:
        data = infile.read()
        total_length = len(data) - 10
        gap_free_length = len(data.replace("N", "")) - 10
        outfile.write(f"Total length  = {total_length}\tGap-free length  = {gap_free_length}\n")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python script.py <input_fasta>")
    else:
        calculate_base_lengths(sys.argv[1])

File: bin/bases.pl
#!/usr/bin/env perl

open(IN,"$ARGV[0]"); # input FASTA 
open(OUT, ">../../../CURRENT/bases.txt");
$b = join("",<IN>);
@a=split(//,$b);
$b=~ s/N//g;
@a2=split(//,$b);
print OUT "Total length  = ";
print OUT ($#a-10);
print OUT "\tGap-free length  = ";
print OUT ($#a2-10);
print OUT "\n";
close(IN);
close(OUT);
exit;
